test/dotnet-new.IntegrationTests/DotnetNewTestTemplatesTests.cs

```diff
@@ -0,0 +1,259 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections.Immutable;
+
+namespace Microsoft.DotNet.Cli.New.IntegrationTests
+{
+    public class DotnetNewTestTemplatesTests : BaseIntegrationTest
+    {
+        private readonly ITestOutputHelper _log;
+
+        private static readonly ImmutableArray<string> SupportedTargetFrameworks =
+        [
+            "net10.0",
+        ];
+
+        private static readonly (string ProjectTemplateName, string ItemTemplateName, string[] Languages, bool SupportsTestingPlatform)[] AvailableItemTemplates =
+        [
```

Comment:
How will you make sure you add any new ones here that get added in the future? Is this going to be documented somewhere?